filebeat/crawler/crawler.go

```diff
@@ -45,6 +42,15 @@ func (c *Crawler) Start(r *registrar.Registrar) error {
 		}
 	}
 
+	if reloaderConfig.Enabled() {
+		logp.Warn("EXPERIMENTAL feature dynamic configuration reloading is enabled.")
+
+		go func() {
+			c.reloader = prospector.NewProspectorReloader(reloaderConfig, c.out, r)
```

Comment:
I don't think anything bad will happen here, but doing the `c.reloader` assignment outside of the goroutine would be less prone to race conditions.